lib/web_ui/lib/src/engine/canvaskit/fonts.dart

```diff
@@ -141,7 +148,15 @@ class SkiaFontCollection {
     }
 
     final Uint8List bytes = buffer.asUint8List();
-    return _RegisteredFont(bytes, family);
+    SkTypeface? typeface =
+        canvasKit.FontMgr.RefDefault().MakeTypefaceFromData(bytes);
+    if (typeface != null) {
+      return _RegisteredFont(bytes, family, typeface);
+    } else {
+      html.window.console.warn('Failed to load font $family at $url');
+      html.window.console.warn('Verify that $url contains a valid font.');
```

Comment:
Let's add an indirection, e.g. a mutable top-level `void Function() printWarning;` that could be mocked in a test so we can test that we log these warnings.

Reply:
Done. PTAL